log/log.go

```diff
@@ -1,159 +1,45 @@
 package log
 
 import (
-	"fmt"
-	"io"
 	"os"
-	"sync"
 
-	"github.com/ViaQ/logerr/kverrors"
+	"github.com/ViaQ/logerr/v2/internal/sink"
 	"github.com/go-logr/logr"
 )
 
-// ErrUnknownLoggerType is returned when trying to perform a *Logger only function
-// that is incompatible with logr.Logger interface
-var ErrUnknownLoggerType = kverrors.New("unknown error type")
-
-var (
-	defaultOutput  io.Writer = os.Stdout
-	defautLogLevel           = 0
-
-	// logLevel sets the level at which you want logs to be displayed
-	// By default the verbosity is set to 0 and all logs that do not
-	// use V(...) will be printed. To increase logging verbosity
-	logLevel = defautLogLevel
-
-	mtx    sync.RWMutex
-	logger logr.Logger = NewLogger("", os.Stdout, 0, JSONEncoder{})
-)
-
-// Init initializes the logger. This is required to use logging correctly
-// component is the name of the component being used to log messages.
-// Typically this is your application name.
-//
-// keyValuePairs are key/value pairs to be used with all logs in the future
-func Init(component string, keyValuePairs ...interface{}) {
-	InitWithOptions(component, nil, keyValuePairs...)
-}
-
-// MustInit calls Init and panics if it returns an error
-func MustInit(component string, keyValuePairs ...interface{}) {
-	Init(component, keyValuePairs...)
+// NewLogger creates a logger with the provided key value pairs
+func NewLogger(component string, keyValuePairs ...interface{}) logr.Logger {
+	return NewLoggerWithOptions(component, nil, keyValuePairs...)
 }
 
-// InitWithOptions inits the logger with the provided opts
-func InitWithOptions(component string, opts []Option, keyValuePairs ...interface{}) {
-	mtx.Lock()
-	defer mtx.Unlock()
+// NewLoggerWithOptions creates a logger with the provided opts and key value pairs
+func NewLoggerWithOptions(component string, opts *Options, keyValuePairs ...interface{}) logr.Logger {
```

Comment:
What's your reason for switching from the `Option` functions to an `Options` struct? We only have two options so far, but to me the option functions are still an elegant way of setting optional information.